pages/history.tsx

```diff
@@ -167,58 +206,78 @@ const History = () => {
             filteredProposals.length > 0 && (
               <div className="space-y-6">
                 {filteredProposals.map((x, i) => {
-                  return x[0] == -1 ? (
-                    <div
-                      key={(x[1] as any).timestamp}
-                      className="grid h-16 w-full grid-cols-3 items-center gap-8 rounded border-b border-zinc-900 bg-zinc-800 px-6 py-4 text-white lg:grid-cols-4"
-                    >
-                      <span className="justify-self-start font-bold md:ml-11">
-                        <span className="hidden md:block">Received Tez</span>
-                        <span className="md:hidden">Received</span>
-                      </span>
-                      <span className="text-center font-light text-zinc-300 md:min-w-[7rem] md:text-left">
-                        <span className="hidden md:inline">From:</span>{" "}
-                        <Alias address={(x[1] as any).sender.address} />
-                      </span>
-                      <span className="truncate font-light text-zinc-300 md:min-w-[7rem]">
-                        <span className="hidden md:inline">Amount:</span>{" "}
-                        {mutezToTez((x[1] as any).amount)} Tez
-                      </span>
-                      <span className="hidden justify-self-end lg:block">
-                        {new Date((x[1] as any).timestamp).toLocaleDateString()}{" "}
-                        -{" "}
-                        {`${new Date((x[1] as any).timestamp)
-                          .getHours()
-                          .toString()
-                          .padStart(2, "0")}:${new Date((x[1] as any).timestamp)
-                          .getMinutes()
-                          .toString()
-                          .padStart(2, "0")}`}
-                      </span>
-                    </div>
-                  ) : (
-                    <ProposalCard
-                      id={x[0]}
-                      key={x[0]}
-                      metadataRender
-                      status={x[1].ui.status}
-                      date={
-                        !!x[1].og.resolver
-                          ? new Date(x[1].og.resolver.timestamp)
-                          : new Date(x[1].ui.timestamp)
-                      }
-                      activities={x[1].ui.signatures.map(
-                        ({ signer, result }) => ({
-                          hasApproved: result,
-                          signer,
-                        })
-                      )}
-                      content={x[1].ui.content}
-                      proposer={x[1].og.proposer}
-                      resolver={x[1].og.resolver}
-                      walletTokens={walletTokens}
-                    />
-                  );
+                  switch (x[0]) {
+                    case TransferType.MUTEZ:
+                      return (
+                        <div
+                          key={(x[1] as any).timestamp}
```

Comment:
About the `any` in the history, I definitely agree we should get rid of them, but the thing is, there some data spreaded here and there, so it would need a good refactor to remove them all. Tbh I can do it, but as we have other priorities and it's not that much a problem, we can postpone it to later I think. Wdyt?

Reply:
Sure, if it takes effort. Close thread.